modules/issue_tracker/php/endpoints/attachment.class.inc

```diff
@@ -0,0 +1,248 @@
+     */
+    private function _handlePOST(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse POST request body.
+        $values     = $request->getQueryParams();
+        $user       = \NDB_Factory::singleton()->user();
+        $attachment = new \LORIS\issue_tracker\UploadHelper();
+        $response   = $attachment->setupUploading(
+            $user,
+            $_FILES,
+            $values
+        );
+        return new \LORIS\Http\Response\JsonResponse(
+            $response
+        );
+    }
+
+    /**
+     * Handle DELETE requests for attachment
+     * and marks attachment as deleted from issue_tracker.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleDELETE(ServerRequestInterface $request)
+    : ResponseInterface
+    {
+        // Verify User can delete file.
+        if ($this->_deletePermission($request)) {
+            // Parse request body.
+            $values    = $request->getQueryParams();
+            $file_hash = $values['file_hash'];
+            $factory   = \NDB_Factory::singleton();
+            $DB        = $factory->database();
+            $query     = 'UPDATE
+                             issues_attachments
+                          SET
+                             deleted = 1
+                          WHERE
+                             file_hash = :file_hash;';
+            $stmt      = $DB->prepare($query);
+            $stmt->execute(array('file_hash' => $file_hash));
+
+            return new \LORIS\Http\Response\JsonResponse(
+                array('success' => true)
+            );
+        }
+        return new \LORIS\Http\Response\JsonResponse(
+            array('error' => 'invalid permission')
+        );
+    }
+
+    /**
+     * Check if user is able to delete the attachment
+     * from the issue_tracker module.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return bool The permission to delete attachment.
+     */
+    private function _deletePermission(ServerRequestInterface $request)
+    : bool
+    {
+        // Parse request body.
+        $values   = $request->getQueryParams();
+        $user     = \NDB_Factory::singleton()->user();
+        $username = $user->getUsername();
+        $ID       = $values['ID'];
+        $DB       = \NDB_Factory::singleton()->database();
+        $query    = 'SELECT
+                        USER
+                    FROM
+                        issues_attachments
+                    WHERE
+                        ID = :ID
+                        AND USER = :user;';
+        $result   = $DB->pselectOne(
+            $query,
+            array('ID' => $ID, 'user' => $username)
+        );
+        // Check if User shouldn't be able to Delete.
+        if ($user->hasPermission('superuser')) {
```

Comment:
You don't have to check this automatically. `hasPermission` always includes a check for superuser. My comment was about the `empty($result)` part. 

Basically if there are two users, `admin` (superuser) and `john`, and `john` uploads a file, your pselect empty check means that `admin` won't be able to delete `john`'s file because the name `admin` is not in the database.

Reply:
@johnsaigle just making note talked offline.